cyto_dl/api/cyto_dl_model/segmentation_plugin_model.py

```diff
@@ -0,0 +1,115 @@
+from pathlib import Path
+from typing import List, Optional, Tuple, Union
+
+from omegaconf import ListConfig
+
+from cyto_dl.api.cyto_dl_model import CytoDLBaseModel
+from cyto_dl.api.data import *
+
+
+class SegmentationPluginModel(CytoDLBaseModel):
+    """A SegmentationPluginModel handles configuration, training, and prediction using the default
+    segmentation_plugin experiment from CytoDL."""
+
+    def __init__(self, config_filepath: Optional[Path] = None):
+        super().__init__(config_filepath)
+        self._has_split_column = False
+
+    # we currently have an override for ['mode'] in ml-seg, but I can't find top-level 'mode' in the configs,
+    # do we need to support this?
+    def _get_experiment_type(self) -> ExperimentType:
+        return ExperimentType.SEGMENTATION_PLUGIN
+
+    def _set_max_epochs(self, max_epochs: int) -> None:
+        self._set_cfg("trainer.max_epochs", max_epochs)
+
+    def _set_manifest_path(self, manifest_path: Union[str, Path]) -> None:
+        self._set_cfg("data.path", str(manifest_path))
+
+    def _set_output_dir(self, output_dir: Union[str, Path]) -> None:
+        self._set_cfg("paths.output_dir", str(output_dir))
+        # I can't find where work_dir is actually used in cyto_dl, do we need to support this?
+        self._set_cfg("paths.work_dir", str(output_dir))
+
+    # a lot of these keys are duplicated across the im2im experiment types (but not present in top-level
+    # train.yaml or eval.yaml) - should we move these into the top-level configs and move these setters and
+    # getters accordingly?
+    def set_spatial_dims(self, spatial_dims: int) -> None:
+        self._set_cfg("spatial_dims", spatial_dims)
+
+    def get_spatial_dims(self) -> int:
+        return self._get_cfg("spatial_dims")
+
+    def set_input_channel(self, input_channel: int) -> None:
+        self._set_cfg("input_channel", input_channel)
+
+    def get_input_channel(self) -> int:
+        return self._get_cfg("input_channel")
+
+    def set_raw_image_channels(self, channels: int) -> None:
+        self._set_cfg("raw_im_channels", channels)
+
+    def get_raw_image_channels(self) -> int:
```

Comment:
@everyone which of the items that I currently have setters/getters for have reasonable defaults? Which should we require the client to choose a value for?

spatial_dims
input_channel
raw_image_channels
patch_size

are the ones I'm not sure about

Reply:
I think we can default input_channel: 0, raw_image_channels:1 and the patch_size set for the smallest model. Spatial dims should be required

Reply:
Follow up question: suppose you train a model with a set of overrides. Which overrides would a user possibly need to make at prediction time? They would need to override output path, checkpoint, etc., but is it ever the case that they would need to override one of the items I listed above? It seems like if you were to train with a certain `patch_size`, you would probably want to predict with that same `patch_size` for instance

Reply:
yeah! patch_size and spatial_dims should be the same, the others might change

Reply:
Good point--and I would also assume that we wouldn't want to change `patch_size`, `spatial_dims`, or `raw_image_channels` when we choose to continue training an existing model? If that's the case, we will need to change the GUI a bit (I'll make a ticket on segmenter)

Reply:
yeah,  I think that's correct